docs/source/migrating-from-gateway.mdx

```diff
@@ -6,20 +6,86 @@ description: An overview of the considerations to make when migrating from Apoll
+
+It'll be good to make a check-list, and you'll want to start by looking for configuration and customizations these places:
+
+- Environment variables
+- Non-Apollo telemetry and instrumentation (i.e., OpenTelemetry, Datadog)
+- Constructor options passed to `new ApolloGateway({ ... })`
+- Constructor options passed to `new ApolloServer({ ... })` (TODO — is this one important? Is plugins most important?)
+- Specific `plugins` passed to `new ApolloServer({ plugins: [ ... ] })`
+- Custom middleware (e.g., Express, Koa, Fastify)
+
+The sections below provide more details on what to look for in each of these categories.
+
+### Environment variables
+
+Environment variables have implicit effect on some Apollo features, like Managed Federation and usage reporting.  Take note of any `APOLLO_` prefixed environment variables that are being set when starting the host runtime you run your Apollo Gateway and Apollo Server on.
+
+All official Apollo-related environment variables are prefixed with `APOLLO_`.  Common cases are:
+
+- `APOLLO_KEY`
+- `APOLLO_GRAPH_REF`
+- TODO More?
+
+In most cases, the Apollo Router supports the same environment variables, though the following are not yet supported:
+
+- `TODO` (probably customizing the uplink endpoints)
+
+## Constructor options passed to `new ApolloGateway({ ... })`
+
+TODO: The options that you pass to the `ApolloGateway` constructor.  (e.g., `supergraphSdl`, `serviceList`, etc.)
+
+### `supergraphSdl`
+
+Generally speaking, the free Apollo managed federation features can be used as the source of truth for Apollo Router, needing only to have the `APOLLO_KEY` environment specified.
+
+When starting it, the Apollo Router takes a `--supergraph` argument which specifies the supergraph file which the Router should use.  This file is watched and the Router's supergraph will hot-reload when the file contents are changed.  Any external process can be used to update the Supergraph, though the most common is `rover supergraph compose` TODO LINK.
+
+### `serviceList`
+
+This method of doing in-process composition was deprecated with the introduction of `supergraphSdl`.  Today, `rover supergraph compose` TODO LINK can can process subgraphs from combination of introspection, SDL files and even Studio-registered variants and turn them into a supergraph.  The supergraph can be passed to the Router using the `--supergraph` argument, and is hot-reloaded on each change.  
+
+### `buildService`
+
+- This is often used to override subgraph URLs at runtime.  Now you can use `override_subgraph_urls`. 
+- This function can return `RemoteGraphQLDataSource` classes which allow changing requests to subgraphs
+  These are often used to change headers that are passed to subgraphs.  This can be done with `headers`.
+  - `willSendRequest`
+  - `didReceiveResponse`
+  - `didEncounterError`
+- `logger`
+
+## Specific `plugins` passed to `new ApolloServer({ plugins: [ ... ] })`
+
+TODO
```

Comment:
Let's link to the plugins stuff,

Also call out explicitly header propagation and the read only plugin